x-pack/plugins/apm/server/lib/services/get_service_dependencies/find_common_connections.test.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SPAN_DESTINATION_SERVICE_RESOURCE } from '../../../../common/elasticsearch_fieldnames';
+import { findCommonConnections } from './find_common_connections';
+import { Connections } from './get_connections';
+
+describe('findCommonConnections', () => {
+  it("doesn't return connections when current and previous are empty", () => {
+    const currentPeriodConnections = [] as Connections;
+    const previousPeriodConnections = [] as Connections;
+
+    expect(
+      findCommonConnections({
+        currentPeriodConnections,
+        previousPeriodConnections,
+      })
+    ).toEqual([]);
+  });
+  it("doesn't return connections when current is empty", () => {
+    const currentPeriodConnections = [] as Connections;
+    const previousPeriodConnections = [
+      {
+        [SPAN_DESTINATION_SERVICE_RESOURCE]: 'opbeans:3000',
+        service: { name: 'opbeans-ruby' },
+      },
+      {
+        [SPAN_DESTINATION_SERVICE_RESOURCE]: 'opbeans:3000',
+        service: { name: 'opbeans-java' },
+      },
+    ] as Connections;
+
+    expect(
+      findCommonConnections({
+        currentPeriodConnections,
+        previousPeriodConnections,
+      })
+    ).toEqual([]);
+  });
+  it("doesn't return connections when there are no match", () => {
+    const currentPeriodConnections = [
+      { [SPAN_DESTINATION_SERVICE_RESOURCE]: 'postgresql' },
+      {
+        [SPAN_DESTINATION_SERVICE_RESOURCE]: 'opbeans:3000',
+        service: { name: 'opbeans-python' },
+      },
+      {
+        [SPAN_DESTINATION_SERVICE_RESOURCE]: 'opbeans:3000',
+        service: { name: 'opbeans-dotnet' },
+      },
+    ] as Connections;
+    const previousPeriodConnections = [
+      {
+        [SPAN_DESTINATION_SERVICE_RESOURCE]: 'opbeans:3000',
+        service: { name: 'opbeans-ruby' },
+      },
+      {
+        [SPAN_DESTINATION_SERVICE_RESOURCE]: 'opbeans:3000',
+        service: { name: 'opbeans-java' },
+      },
+    ] as Connections;
+
+    expect(
+      findCommonConnections({
+        currentPeriodConnections,
+        previousPeriodConnections,
+      })
```

Comment:
Does this mean that if the dependencies found for the current period are not found in the previous period, nothing is displayed?

Reply:
No, the dependencies for the current period will always be displayed even if none is found in the previous period. Maybe a better name for this function would be `findPreviousPeriodDependencies`. WDYT?

Reply:
That sounds better, but I'm wondering if this can be simplified. E.g., merge data on the server, and use the connections/metrics from the current period as what you're iterating over, then for every connection, look for metrics from the previous period that have the same span.destination.service.resource.

Reply:
> look for metrics from the previous period that have the same span.destination.service.resource.

Should I look only for the `span.destination.service.resource` property? 

Because there're cases when the current period has `span.destination.service.resource=opbeans:3000` and `service.name=opbeans-java` and the previous period has `span.destination.service.resource=opbeans:3000` but `service.name=opbeans-node`, should I show the previous period in that case?

Reply:
okay, I got it.